theseus/optimizer/nonlinear/gauss_newton.py

```diff
@@ -41,3 +41,29 @@ def __init__(
 
     def compute_delta(self, **kwargs) -> torch.Tensor:
         return self.linear_solver.solve()
+
+    def compute_samples(self, n_samples: int = 10, T: float = 1.0) -> torch.Tensor:
+        delta = self.linear_solver.solve()
+        AtA = self.linear_solver.precision() / T
+        R = torch.linalg.cholesky(AtA, upper=True)
+
+        B, N = delta.shape
+        y = torch.normal(
+            mean=torch.zeros((N, n_samples), device=delta.device),
+            std=torch.ones((N, n_samples), device=delta.device),
+        )
+        delta_samples = (torch.triangular_solve(y, R).solution) + (
```

Comment:
Add a comment for reference to the math.

Reply:
Added a reference.